### PR TITLE
New channel permissions helper

### PR DESCRIPTION
**Pull request by zuuring**
### Notes
This commit adds a new `channel-management` helper to Valkyrie which will always ensure permissions on new channels within the `defense` category are assigned to the same permission set as the base channel category permissions.

It then sends a message to the newly created channel, listing the permissions and access levels they have for quick verification on channel permissions.

If the channel is created with different permissions than base category, it will be overwritten by the permissions on base category. However, permissions can also be altered after creation if needed manually. 

### Screenshot
Screenshot 2024-03-05 at 11 04 56

### CODE REVIEW

**Comment by zuuring** [comment]
Also marking this one ready to review, if we want to get the permission mapper for #defense cat pushed before our next release.

**Review by Shadowfiend — APPROVED** [review]
This is pretty much good to go.

I will leave it up to you if you want to merge & then address in one of the following PRs, or if you want to address & then we merge.

**Comment by zuuring** [comment]
@Shadowfiend Thanks for the review comments and have now fixed this up so it's ready to be merged. Just need to add env `DEFENSE_CATEGORY_ID` beforehand.

File: discord-scripts/role-management.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Client, TextChannel } from "discord.js"
+import { Robot } from "hubot"
+
+export default async function manageChannelPermissions(
+  discordClient: Client,
+  robot: Robot,
+) {
+  const { application } = discordClient
+  if (process.env.DEFENSE_CATEGORY_ID) {
+    if (application) {
```
Comment (Shadowfiend) [inline discord-scripts/role-management.ts:10]: No need to nest, these can combine into one.

**Review by Shadowfiend — APPROVED** [review]
Pushed a commit with the kube config updates. Let's do it.

**Pull request merged**